converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/FhirTransaction/Observation/ObservationParser.cs

```diff
@@ -0,0 +1,343 @@
+                    (string Value, string Scheme) lookupTuple = (report.Code.Value, report.Code.Scheme);
+                    if (ObservationConstants.DoseSummaryReportCodes.Contains(lookupTuple))
+                    {
+                        Observation doseSummary = CreateDoseSummary(
+                            dataset,
+                            imagingStudyRef,
+                            patientRef);
+                        observations.DoseSummaries.Add(doseSummary);
+                    }
+
+                    if (ObservationConstants.IrradiationEventCodes.Contains(lookupTuple))
+                    {
+                        Observation irradiationEvent = CreateIrradiationEvent(dataset, patientRef);
+                        observations.IrradiationEvents.Add(irradiationEvent);
+                    }
+                }
+            }
+            catch (MissingMemberException)
+            {
+                // Occurs when a required attribute is unable to be extracted from a dataset.
+                // Ignore and move onto the next one.
+            }
+
+            // Recursively iterate through every child in the document checking for nested observations.
+            // Return the final aggregated list of observations.
+            foreach (DicomContentItem childItem in report.Children())
+            {
+                ObservationParseLoop(childItem.Dataset, patientRef, imagingStudyRef, observations);
+            }
+        }
+
+        /// <summary>
+        /// Creates an Observation "Dose Summary" profile based on the the spec outlined at:
+        /// https://confluence.hl7.org/display/IMIN/Radiation+Dose+Summary+for+Diagnostic+Procedures+on+FHIR#RadiationDoseSummaryforDiagnosticProceduresonFHIR-DoseSummaryObservationProfile
+        /// </summary>
+        /// <returns>A Dose Summary</returns>
+        private static Observation CreateDoseSummary(
+            DicomDataset dataset,
+            ResourceReference imagingStudyRef,
+            ResourceReference patientRef)
+        {
+            // Create the observation
+            var observation = new Observation
+            {
+                // Set the code.coding
+                Code = new CodeableConcept("http://loinc.org", "73569-6", "Radiation exposure and protection information"),
+                // Add Patient reference
+                Subject = patientRef
+            };
+            // Add ImagingStudy reference
+            observation.PartOf.Add(imagingStudyRef);
+
+            // Set identifiers
+            if (dataset.TryGetSingleValue(DicomTag.StudyInstanceUID, out string studyUid))
+            {
+                Identifier identifier = ImagingStudyIdentifierUtility.CreateIdentifier(studyUid);
+                observation.Identifier.Add(identifier);
+            }
+            else
+            {
+                throw new MissingMemberException($"Unable to {nameof(DicomTag.StudyInstanceUID)} from dose summary observation dataset");
```

Comment:
Is this exception from fo-dicom? We should not throw fo-dicom exceptions. We should created our own.
Also please follow the coding standard. The exceptions messages should be in resx file. Also don't use dollar format when logging, it is less performant

Reply:
This is not thrown by fo-dicom -- it is part of [dotnet std lib](https://docs.microsoft.com/en-us/dotnet/api/system.missingmemberexception?view=net-5.0). I used it because it conveyed the fact that a property we expect a DicomDataset to contain is missing.